### PR TITLE
front end setup

### PR DESCRIPTION
**Pull request by nourishnew**
Clone the repository. Go to the client directory through the terminal and do "npm install" to set up all the node modules. "npm start" launches the app

### CODE REVIEW

File: can_explorer_client/.eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"D:\\Waterloo Computer Engineering\\2AMidnightsun\\CAN-Explorer\\can_explorer_client\\src\\index.js":"1","D:\\Waterloo Computer Engineering\\2AMidnightsun\\CAN-Explorer\\can_explorer_client\\src\\reportWebVitals.js":"2","D:\\Waterloo Computer Engineering\\2AMidnightsun\\CAN-Explorer\\can_explorer_client\\src\\App.js":"3"},{"size":500,"mtime":499162500000,"results":"4","hashOfConfig":"5"},{"size":362,"mtime":499162500000,"results":"6","hashOfConfig":"5"},{"size":140,"mtime":1611431956708,"results":"7","hashOfConfig":"5"},{"filePath":"8","messages":"9","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"e18iux",{"filePath":"10","messages":"11","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"12","messages":"13","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"D:\\Waterloo Computer Engineering\\2AMidnightsun\\CAN-Explorer\\can_explorer_client\\src\\index.js",[],"D:\\Waterloo Computer Engineering\\2AMidnightsun\\CAN-Explorer\\can_explorer_client\\src\\reportWebVitals.js",[],"D:\\Waterloo Computer Engineering\\2AMidnightsun\\CAN-Explorer\\can_explorer_client\\src\\App.js",[]]
```
Comment (AveryChiu64) [inline can_explorer_client/.eslintcache]: Looks like there are some unneeded files from the cache here, could these be removed.

File: can_explorer_client/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
+
+Runs the app in the development mode.\
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
```
Comment (AveryChiu64) [inline can_explorer_client/README.md:12]: I think Raiyan said he was forwarding port 8000 from the host to the VM, not 3000

File: can_explorer_client/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+**Note: this is a one-way operation. Once you `eject`, you can’t go back!**
+
+If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+
+Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
+
+You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
+
+## Learn More
```
Comment (AveryChiu64) [inline can_explorer_client/README.md:42]: A lot of this seems like cookie-cutter stuff. Can we remove this and only leave out the things that are necessary for the user for the software?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,14 @@
...
 An application to visualize CAN data and interact with the CAN bus
+
+Clone the repository :
+git clone "url"
+
+Make a branch :
+git checkout -b <branch_name>
+
+Install node modules :
+npm install
```
Comment (AveryChiu64) [inline README.md:23]: I don't think our VM has npm installed by default (at least I don't). Consider adding instructions at least a link to something like this here: https://github.com/nodesource/distributions

Comment (AveryChiu64) [reply README.md:23]: I'm also not sure, what verison you're using? Probably best to write that down too.

**Review by AveryChiu64 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (AveryChiu64) [reply README.md:23]: In my case, I had to run
```
curl -sL https://deb.nodesource.com/setup_15.x | sudo -E bash -
sudo apt-get install -y nodejs
```

File: can_explorer_client/README.md (hunk 2 above)
Comment (raiyansayeed) [reply can_explorer_client/README.md:12]: If we want to run a Django + React app, it looks like we will need two ports anyway (one running the backend and another running the React front-end). This [guide](https://www.digitalocean.com/community/tutorials/build-a-to-do-application-using-django-and-react) goes into the idea in more detail.  We would have to add another port-forwarding rule in the Vagrantfile

Comment (AveryChiu64) [reply can_explorer_client/README.md:12]: Sure, Raiyan could you add this to the Vagrantfile then? I'm fine with exposing both port 8000 and 3000.

**Comment by AveryChiu64** [comment]
Closed since this is outdated

**Pull request closed without merging**